test/test_parameter.h

```diff
@@ -62,6 +62,38 @@ TEST(parameter, layer_adder) {
   EXPECT_EQ(parameters[1]->type(), parameter_type::bias);
 }
 
+TEST(parameter, xavier_init) {
+  set_random_seed(42u);
+
+  parameter_init::xavier xvr;
+  Parameter parameter(1, 1, 1, 10, parameter_type::weight);
+  xvr.fill(&parameter, 1, 1);
```

Comment:
@karandesai-96 thinking about design, it makes more sense to have a member function `initialize(init_fcn)` in Parameter that receives the function.